### PR TITLE
Apex charts

### PR DESCRIPTION
**Pull request by 100starnight**
Resolves #

## Proposed Changes

-

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/components/navi-visualizations/apex-gauge.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+export default class NaviVisualizationsApexGauge extends Component {
+  layout = layout;
+
+  /**
+   * @property {number} - the current value of the metric being displayed
+   */
+  @computed('model')
+  get data() {
+    const metric = this.model.firstObject.request.metrics['0'].metric;
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-gauge.js:14]:
```suggestion
```

File: packages/core/addon/components/navi-visualizations/apex-gauge.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import Component from '@ember/component';
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-gauge.js:1]: Remember to include the copyright statement for addon files

File: packages/core/addon/components/navi-visualizations/apex-gauge.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  /**
+   * @property {string} - the hex color of the gauge
+   */
+  @computed('series')
+  get gaugeColor() {
+    // thresholds:
+    // red < 75 <= yellow <= 85 < green
+    if (this.series[0] < 75) {
+      return '#f05050';
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-gauge.js:38]: Can we declare these color codes as consts and get rid of the comment?
`const RED = '#f05050';` etc.

File: packages/core/addon/components/navi-visualizations/apex-pie.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import layout from '../../templates/components/navi-visualizations/apex-pie';
+import { alias } from '@ember/object/computed';
+export default class NaviVisualizationsApexPie extends Component {
+  layout = layout;
+  @service() bardMetadata;
+
+  @alias('options.metadata.series.config.metric.metric') metric;
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-pie.js:10]:
```suggestion
  get metric() {
    return this.options.metadata.series.config.metric.metric;
  }
```
Let's avoid creating two-way data bindings

File: packages/core/addon/components/navi-visualizations/apex-pie.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    val = val.id + '|' + val.descriptionTag;
+    return val;
+  }
+
+  /**
+   * @property {object} - an ApexCharts-friendly object of the data and data labels
+   */
+  @computed('model')
+  get data() {
+    let vals = { series: [], labels: [] };
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-pie.js:31]:
```suggestion
    const vals = { series: [], labels: [] };
```

File: packages/core/addon/components/navi-visualizations/apex-pie.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    return vals;
+  }
+
+  /**
+   * @property {object} - ApexCharts-compatible object of options
+   */
+  @computed('series', 'labels')
+  get chartOptions() {
+    let pie = {
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-pie.js:45]:
```suggestion
    return {
```

File: packages/core/addon/components/navi-visualizations/apex-pie.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      chart: { type: 'pie' },
+      series: this.data.series,
+      labels: this.data.labels
+    };
+    return pie;
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/apex-pie.js:50]:
```suggestion
```

File: packages/core/tests/integration/components/navi-visualizations/apex-gauge-test.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        options=(hash
+          baselineValue=0
+          goalValue=100000000000
+          metric=metric
+        )
+      }}
+    `);
+    // checks correct color: red = #f05050 = rgba(240, 80, 80, X)
+    assert.equal(
+      this.element.querySelector('.apexcharts-radial-series path').getAttribute('stroke'),
```
Comment (mattgraham1995) [inline packages/core/tests/integration/components/navi-visualizations/apex-gauge-test.js:27]: Can we use QUnit DOM assertions where possible?
https://github.com/simplabs/qunit-dom

File: packages/core/addon/components/navi-visualizations/apex-gauge.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import numeral from 'numeral';
+import layout from '../../templates/components/navi-visualizations/apex-gauge';
+
+export default class NaviVisualizationsApexGauge extends Component {
+  layout = layout;
```
Comment (jkusa) [inline packages/core/addon/components/navi-visualizations/apex-gauge.js:7]: Let's use glimmer components for all the new visualizations.

**Comment by jkusa** [comment]
Can we name the integration branch something like `apex-charts`

**Pull request closed without merging**